src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerEvents.cs

```diff
@@ -69,51 +164,70 @@ private void Source_PackageInstalled(object sender, PackageEventArgs e)
             NotifyInstalled(new PackageOperationEventArgs(e.InstallPath, e.Identity, null));
         }
 
+        private void NotifyDelegates(PackageOperationEventArgs e, Delegate[] delegates)
+        {
+            VsPackageMetadata eventData = new(e.Package, e.InstallPath);
+            for (int i = 0; i < delegates.Length; i++)
+            {
+#pragma warning disable CA1031 // Do not catch general exception types
```

Comment:
Why disable this warning instead of just using `catch (Exception)`?

Reply:
CA1031 is still raised.

All exceptions extend `Exception`, hence `catch (Exception) {}` and `catch {}` are functionally equivalent.

Reply:
Ah sorry I'm used to ReSharper which warns with `catch {}` but not `catch (Exception) {}` and the docs say this so I thought it was saying the same thing, at least catch something:

> Do not suppress a warning from this rule. Catching general exception types can hide run-time problems from the library user and can make debugging more difficult.

Too bad there's no way to convey that you actually know what you're doing by ignoring any exceptions without just suppressing the warning...